pkg/ingester/activeseries/active_series.go

```diff
@@ -472,7 +472,7 @@ func (s *seriesStripe) purge(keepUntil time.Time, idx tsdb.IndexReader) {
 				if err := idx.Series(ref, &buf, nil); err != nil {
 					s.activeSeriesAttributionFailureCounter.Add(1)
 				} else {
-					s.cat.Decrement(buf.Labels())
+					s.cat.Decrement(buf.Labels(), entry.numNativeHistogramBuckets)
```

Comment:
I see that in `findAndUpdateOrCreateEntryForSeries` we can modify `entry.numNativeHistogramBuckets`, which means that this number isn't necessarily the same we called on `s.cat.Increment(series, time.Unix(0, nowNanos), numNativeHistogramBuckets)` above, which means we could go negative values (or have leftovers).

Please add a test that tracks a series with given labels, then that series turns into a native histogram, and then it's removed, and check that cost attribution tracker is at zero.

Reply:
in commit fe7acada914ff9ac74f5992afeb86f857a54ba83, i added a test to decrement more on bucket than incremented, the bucket counter should stay at 0, also when number of native histogram series become 0, the total bucket should be 0

Reply:
addressed accordingly in commit bb0a5dcb405890615156fecaaa44b4c003e4d7a1